src/Microsoft.OpenApi.OData.Reader/Edm/ODataOperationSegment.cs

```diff
@@ -130,14 +131,17 @@ private string FunctionName(IEdmFunction function, OpenApiConvertSettings settin
                 }
             }
 
-            StringBuilder functionName = new StringBuilder();
+            StringBuilder functionName = new();
+            string selectedName = function.FullName();
+
             if (settings.EnableUnqualifiedCall)
             {
-                functionName.Append(function.Name);
+                selectedName = settings.DefaultNamespace != null ? selectedName.RemoveBaseNamespace(settings) : function.Name;
```

Comment:
move the ternary logic into the method so we don't end up duplicating it all over the place.
check with string.isnullorempty so this doesn't fail if we enable NRT one day.